### DISCUSSION TITLE
Add a Device to track

### DISCUSSION THREAD

**Discussion by andersstigel** [leolabs/ableton-js#122] [Q&A] [answered]
Thanks for adding so many new and cool features 🙏

Do you know if it is possible to add a specific device to the device chain of a track?

For example adding: 

- the default wavetable, 
- a wavetable preset, 
- the default drum rack
- audio effects
- etc

I see the live object model has a delete_device function but no add_device? 

Thank you 🚀

**Comment by leolabs** [comment, marked as answer]
I'm afraid this isn't currently possible with Live's API. I've also taken a look at the available functions but couldn't find any way to programmatically add a given device to a track.

**Comment by andersstigel** [comment]
Thanks! Maybe we will try looking into a work around: having a project with a lot tracks with different instruments and devices, and then programatically remove unessescary ones.
